### ISSUE TITLE
[ENHANCEMENT] Support list and dictionary data structures

### ISSUE THREAD

**Issue by tomhamer** [marqo-ai/marqo#277]
**Is your feature request related to a problem? Please describe.**
Users of Marqo cannot use list or dictionary data structures as attributes. This is a limitation as there is no good way to do a "contains" operation on a metadata field when filtering. It also restricts the user from storing more kinds of metadata in Marqo.

**Describe the solution you'd like**
Users should be able to use list or dictionary data structures as attributes within fields. 

**Describe alternatives you've considered**
Another alternative would be to allow users to store lexical fields as metadata within chunks. This would increase the amount of storage required but would allow users to do "contains" operations in filtering at the very least through query strings.

**Comment by pandu-k** [comment]
Non-tensor lists have been implemented: https://github.com/marqo-ai/marqo/pull/312
